### ISSUE TITLE
Cannot read `timeout` of undefined in spark.js

### ISSUE THREAD

**Issue by 3rd-Eden** [primus/primus#387]
Under certain circumstances it's possible that the `spark.js` file throws an error because `spark.primus` is undefined. This seems to happen when server is starting and crashing after a while as I could only see occurrences of this happening during a crash loop of my application.

The only thing my application code is doing is a `primus.forEach` with a `primus.emit` when new data is received as shown in: https://github.com/DestinyTheGame/gorgon.exchange/blob/master/index.js#L116-L118

```
/root/gorgon.exchange/node_modules/primus/spark.js:109
  if ('number' !== typeof spark.primus.timeout) return spark;
                                      ^
TypeError: Cannot read property 'timeout' of undefined
    at Sparky.heartbeat (/root/gorgon.exchange/node_modules/primus/spark.js:109:39)
    at WebSocket.emit (events.js:110:17)
    at Receiver.onping (/root/gorgon.exchange/node_modules/ws/lib/WebSocket.js:816:10)
    at /root/gorgon.exchange/node_modules/ws/lib/Receiver.js:645:14
    at Receiver.flush (/root/gorgon.exchange/node_modules/ws/lib/Receiver.js:336:3)
    at Receiver.opcodes.9.finish (/root/gorgon.exchange/node_modules/ws/lib/Receiver.js:648:12)
    at Receiver.expectHandler (/root/gorgon.exchange/node_modules/ws/lib/Receiver.js:630:33)
    at Receiver.add (/root/gorgon.exchange/node_modules/ws/lib/Receiver.js:91:24)
    at Socket.firstHandler (/root/gorgon.exchange/node_modules/ws/lib/WebSocket.js:775:22)
    at Socket.emit (events.js:107:17)
```

**Comment by lpinca** [comment]
I think that this has already been fixed in https://github.com/primus/primus/pull/367 which is available in `primus@^3.1.0`.

**Comment by 3rd-Eden** [comment]
That could indeed be the fix here.
